sonar-slang-plugin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -214,12 +218,27 @@ private Tree convertElementToSlangAST(PsiElement element, TreeMetaData metaData)
   }
 
   private Tree createClassDeclarationTree(TreeMetaData metaData, KtClass ktClass) {
-    Tree classDecl = createNativeTree(metaData, new KotlinNativeKind(ktClass, ktClass.getName()), ktClass);
-    return new ClassDeclarationTreeImpl(metaData, classDecl);
+    PsiElement nameIdentifier = ktClass.getNameIdentifier();
+    IdentifierTree identifier = null;
+    String name = null;
+
+    if (nameIdentifier != null) {
+      name = nameIdentifier.getText();
+      identifier = createIdentifierTree(getTreeMetaData(nameIdentifier), name);
+    }
+    Boolean isInsideEnum = enumClassDeclaration.peekLast();
+    enumClassDeclaration.addLast(ktClass.isEnum());
+    Tree nativeClassDecl = createNativeTree(metaData, new KotlinNativeKind(ktClass, name), ktClass);
```

Comment:
Why do we need to have the name in the native kind?

Reply:
Good point. I tried removing it, but then equivalence test were failing. It seems for some reasons the `KtNameIdentifier` is not in the children of the `KtClass` object...
I fixed our converter to solve that problem in 8cddace7a47c66a5dde9c1cf14cf3da4e1d5e0e6